File: DSA/Ananya_Gambhir/Day15_ques1.java
import java.util.Scanner;
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}
public class Day15_ques1 {
    public static int height(TreeNode root) {
        if (root == null) {
            return -1;  
        }
        int leftHeight = height(root.left);
        int rightHeight = height(root.right);

        return Math.max(leftHeight, rightHeight) + 1;
    }

    public static TreeNode buildTree(Scanner scanner) {
        System.out.print("Enter node value (or 'null' for no node): ");
        String input = scanner.next();

        if (input.equals("null")) {
            return null;
        }

        int value = Integer.parseInt(input);
        TreeNode node = new TreeNode(value);

        System.out.println("Entering left child of " + value);
        node.left = buildTree(scanner);

        System.out.println("Entering right child of " + value);
        node.right = buildTree(scanner);
        return node;
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Building the binary tree:");
        TreeNode root = buildTree(scanner);

        int treeHeight = height(root);
        System.out.println("The height of the binary tree is: " + treeHeight);

        scanner.close();
    }
}

File: DSA/Anisha_Parveen/Day_15_Ques_1.cpp
#include <iostream>
using namespace std;

struct TreeNode {
    int value;
    TreeNode* left;
    TreeNode* right;

    TreeNode(int val) : value(val), left(nullptr), right(nullptr) {}
};

int heightOfBinaryTree(TreeNode* root) {
    if (root == nullptr) {
        return -1;
    }

    int leftHeight = heightOfBinaryTree(root->left);
    int rightHeight = heightOfBinaryTree(root->right);

    return max(leftHeight, rightHeight) + 1;
}

int main() {
    TreeNode* root = new TreeNode(3);
    root->left = new TreeNode(2);
    root->right = new TreeNode(5);
    root->left->left = new TreeNode(1);
    root->right->left = new TreeNode(4);
    root->right->right = new TreeNode(6);

    int height = heightOfBinaryTree(root);

    cout << "The height of the binary tree is: " << height << endl;

    return 0;
}
